### PR TITLE
IGNITE-15629 Snapshot command implemented

### PR DESCRIPTION
**Pull request by nizhikov**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/management/api/Command.java [hunk 1]
```diff
@@ -57,8 +57,11 @@ public default boolean experimental() {
         return null;
     }
 
-    /** @return Deprecation message if command on the way to be decomissioned. */
-    public default @Nullable String deprecationMessage() {
+    /**
+     * @param arg Command argument.
+     * @return Deprecation message if command on the way to be decomissioned.
```
[the remark this answers is not in the document]
Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/internal/management/api/Command.java]: Thanks, fixed.

File: modules/core/src/main/java/org/apache/ignite/internal/management/snapshot/SnapshotCreateCommandArg.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+/** */
+public class SnapshotCreateCommandArg extends IgniteDataTransferObject {
+    /** */
+    private static final long serialVersionUID = 0;
+
+    /** */
+    @Positional
+    @Argument(description = "Snapshot name. " +
+        "In case incremental snapshot (--incremental) full snapshot name must be provided")
```
Comment (ivandasch) [inline modules/core/src/main/java/org/apache/ignite/internal/management/snapshot/SnapshotCreateCommandArg.java:36]: In the case of

File: modules/core/src/main/java/org/apache/ignite/internal/management/snapshot/SnapshotRestoreCommandArg.java [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+ */
+@ArgumentGroup(value = {"start", "status", "cancel"}, onlyOneOf = true, optional = true)
+public class SnapshotRestoreCommandArg extends IgniteDataTransferObject {
+    /** */
+    private static final long serialVersionUID = 0;
+
+    /** */
+    @Positional
+    @Argument(description = "Snapshot name. " +
+        "In case incremental snapshot (--incremental) full snapshot name must be provided")
```
Comment (ivandasch) [inline modules/core/src/main/java/org/apache/ignite/internal/management/snapshot/SnapshotRestoreCommandArg.java:39]: In the case of

File: modules/core/src/main/java/org/apache/ignite/internal/management/snapshot/SnapshotCreateCommandArg.java (hunk 2 above)
Comment (ivandasch) [inline modules/core/src/main/java/org/apache/ignite/internal/management/snapshot/SnapshotCreateCommandArg.java:36]: In case of

**Review by ivandasch — APPROVED** [review]

**Pull request merged**